sources/api/apiserver/src/server/mod.rs

```diff
@@ -379,6 +424,38 @@ async fn reboot() -> Result<HttpResponse> {
 
 // Helpers for handler methods called by the router
 
+fn call_thar_be_updates(args: Vec<String>) -> Result<HttpResponse> {
+    let status = Command::new("/usr/bin/thar-be-updates")
+        .args(args)
+        .status()
+        .context(error::UpdateDispatcher)?;
+    if status.success() {
+        Ok(HttpResponse::NoContent().finish())
+    } else {
+        let exit_status = match status.code() {
+            Some(code) => code,
+            None => status.signal().unwrap_or(1),
+        };
+        Err(match_update_dispatcher_errors(exit_status, None))
```

Comment:
I'd recommend joining these functions; this sounds generic, but it's specific to t-b-u error codes.  And both are short.